### ISSUE TITLE
MD032: warning / fix only for top line

### ISSUE THREAD

**Issue by skwde** [DavidAnson/vscode-markdownlint#300]
Following pic shows the problem:

<img width="707" alt="Screenshot 2023-09-13 at 10 22 33" src="https://github.com/DavidAnson/vscode-markdownlint/assets/26454702/bb50b274-71d9-4669-9ae6-4cd135e6dd1e">

MD032 only shows up for the line above a list and ignores the one below.
It also only adds a blank line after `Some text`

According to https://github.com/DavidAnson/markdownlint/blob/main/doc/md032.md a blank line should be added before **and** after list items.

**Comment by DavidAnson** [comment]
According to the CommonMark specification, the text on line 11 of your image is a continuation of the list item above it and therefore a blank line should not be added because it would change the content of the list item. The specification calls this a "lazy continuation line" and you can see in the demo page that it's part of the second list item: https://dlaa.me/markdownlint/#%25m%23%20Issue%20300%0A%0A1.%20Some%0A2.%20List%0ASome%20text%0A

**Comment by skwde** [comment]
Oh yes, you are of course right.

In that case I think that MD032 is confusing because it clearly states at https://github.com/DavidAnson/markdownlint/blob/main/doc/md032.md:

> MD032 - Lists should be surrounded by blank lines

This is also highlighted through the example code and how it *should* be fixed.
But the linting is actually behaving differently.

Maybe an option can be added to opt-out lazy continuation line.

Not sure though if also something like this 

```markdown
1. abc
2. def
   ghi

```

also counts as laze continuation line, i.e. when spaces are added before the text on the lazy continuation line.

**Comment by DavidAnson** [comment]
Your latest example is a standard list (not lazy). I agree the example in the documentation is wrong and will use this issue to fix that, thank you.
